h2/utilities.py

```diff
@@ -197,10 +197,10 @@ def validate_headers(headers, hdr_validation_flags):
     headers = _reject_surrounding_whitespace(
         headers, hdr_validation_flags
     )
-    headers = _reject_te(
+    headers = _strip_connection_header(
```

Comment:
I suspect the answer should be that we should consider this a normalisation step on outbound headers only. We should continue to hold our peers to the same standard as RFC7540 requires, but we can make our users lives easier by stripping these headers for them. That means this should be a new normalisation step in `normalize_outbound_headers`, rather than an adjustment of this current filtering logic.